holidays/countries/qatar.py

```diff
@@ -0,0 +1,168 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see AUTHORS.md file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from gettext import gettext as tr
+
+from holidays.calendars import _CustomIslamicHolidays
+from holidays.calendars.gregorian import (
+    JAN,
+    MAR,
+    APR,
+    MAY,
+    JUN,
+    JUL,
+    AUG,
+    SEP,
+    OCT,
+    NOV,
+    DEC,
+    FRI,
+    SAT,
+)
+from holidays.constants import BANK, PUBLIC
+from holidays.groups import InternationalHolidays, IslamicHolidays, StaticHolidays
+from holidays.holiday_base import HolidayBase
+
+
+class Qatar(HolidayBase, InternationalHolidays, IslamicHolidays, StaticHolidays):
+    """Qatar holidays.
+
+    References:
+        * <https://en.wikipedia.org/wiki/Public_holidays_in_Qatar>
+        * [National Sports Day](https://hukoomi.gov.qa/en/national-sport-day)
+        * [Qatar National Day](https://www.qatar.qa/en/qatar/history-of-qatar-qatar-national-day-committee/)
+    """
+
+    country = "QA"
+    default_language = "ar_QA"
+    # %s (estimated).
+    estimated_label = tr("%s (المقدرة)")
+    start_year = 1971
+    supported_categories = (BANK, PUBLIC)
+    supported_languages = ("ar_QA", "en_US")
+    weekend = {FRI, SAT}
+
+    def __init__(self, islamic_show_estimated: bool = True, *args, **kwargs):
+        """
+        Args:
+            islamic_show_estimated:
+                Whether to add "estimated" label to Islamic holidays name
+                if holiday date is estimated.
+        """
+        InternationalHolidays.__init__(self)
+        IslamicHolidays.__init__(
+            self, cls=QatarIslamicHolidays, show_estimated=islamic_show_estimated
+        )
+        StaticHolidays.__init__(self, QatarStaticHolidays)
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        if self._year >= 2012:
```

Comment:
```suggestion
        # Qatar switches from THU-FRI to FRI-SAT on Aug 1, 2003.
        self.weekend = {THU, FRI} if self._year <= 2003 else {FRI, SAT}

        if self._year >= 2012:
```
For `weekend` change like this, weekends are declared in `self.weekend` in `_populate_public_holidays` instead. you can remove the previous `weekend` declaration for `class Qatar` above.

Make sure to update the sources in docstring to include https://www.arabnews.com/node/234601 as well

Reply:
ahh! nice catch! I've updated it along with the references in the docstring.